packages/appsync-modelgen-plugin/src/visitors/appsync-swift-visitor.ts

```diff
@@ -624,13 +624,19 @@ export class AppSyncSwiftVisitor<
           connectionInfo.connectedModel,
         )}.keys.${this.getFieldName(connectionInfo.associatedWith)})`;
       }
-      if (connectionInfo.kind === CodeGenConnectionType.HAS_ONE && (connectionInfo.targetNames || connectionInfo.targetName)) {
-        const targetNameAttrStr = this.isCustomPKEnabled() && connectionInfo.targetNames
-          ? `targetNames: [${connectionInfo.targetNames.map(target => `"${target}"`).join(', ')}]`
-          : `targetName: "${connectionInfo.targetName}"`;
-        return `.hasOne(${name}, is: ${isRequired}, ofType: ${typeName}, associatedWith: ${this.getModelName(
-          connectionInfo.connectedModel,
-        )}.keys.${this.getFieldName(connectionInfo.associatedWith)}, ${targetNameAttrStr})`;
+      if (connectionInfo.kind === CodeGenConnectionType.HAS_ONE) {
+        if (connectionInfo.targetNames || connectionInfo.targetName) {
+          const targetNameAttrStr = this.isCustomPKEnabled() && connectionInfo.targetNames
+            ? `targetNames: [${connectionInfo.targetNames.map(target => `"${target}"`).join(', ')}]`
+            : `targetName: "${connectionInfo.targetName}"`;
+          return `.hasOne(${name}, is: ${isRequired}, ofType: ${typeName}, associatedWith: ${this.getModelName(
+            connectionInfo.connectedModel,
+          )}.keys.${this.getFieldName(connectionInfo.associatedWith)}, ${targetNameAttrStr})`;
+        } else {
+          return `.hasOne(${name}, is: ${isRequired}, ofType: ${typeName}, associatedWith: ${this.getModelName(
+            connectionInfo.connectedModel,
+          )}.keys.${this.getFieldName(connectionInfo.associatedWith)})`;
```

Comment:
I haven't tracked down `getFieldName`, but will this work for an association with a CPK?

```graphql
type Primary @model {
  pk: ID! @primaryKey(sortKeyFields: ["sk1", "sk2"])
  sk1: ID!
  sk2: ID!
  relateds: [Related] @hasMany(references: ["primaryPk", "primarySk1", "primarySk2"])
  content: String
}
type Related @model {
  id: ID!
  content: String
  primaryPk: ID
  primarySk1: ID
  primarySk2: ID
  primary: Primary @belongsTo(references: ["primaryPk", "primarySk1", "primarySk2"])
}
```

Reply:
I've checked with the Swift team and they say that the `associatedWith` is not actually used by the swift client so having only one field should not affect the functionality.

Reply:
After further discussion with the Swift team, the previous statements is incorrect. We will need to make an additional change to codegen.